File: AlgorithmRise/extra/LeetCode/Algorithms/226. Invert Binary Tree/draft-1.c
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */

var swap = function(node_left, node_right){
    let temp;

        temp = node_right.val;
        node_right.val = nodes[0].left.val;
        nodes[0].left.val = temp;
}

var visitTree = function(nodes){

    let length = nodes.length;

    if(0 === length){
        return;
    }

    let temp;
    if(1 === length){
        swap(nodes[0], nodes[1]);
        visitTree([nodes[0].left, nodes[0].right]);
        return;
    }

    let end = length/2;
    for( let i = 0; i<end; ++i ){
        
    }

    visitTree(nodes);
}


/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function(root) {
    
    visitTree(root);

    return root;
};

File: AlgorithmRise/extra/LeetCode/Algorithms/897. Increasing Order Search Tree/draft-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {

    public TreeNode traverseTree(TreeNode p){
        
        if(null == p){
            return null;
        }

        TreeNode node_left = traverseTree(p.left);
        TreeNode node_right = traverseTree(p.right);

        if(null == node_left && null == node_right){
            return p;
        }else if( null != node_left  && null == node_right ){
            node_left.right = p;
        }else if( null == node_left  && null != node_right ){
            return node_right;
        }else{

            return p;
        }

        return p;
    }

    public TreeNode increasingBST(TreeNode root) {
        return traverseTree(root);
    }
}